src/AppInstallerCLITests/Filesystem.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "TestCommon.h"
+#include <winget/Filesystem.h>
+#include <AppInstallerStrings.h>
+
+using namespace AppInstaller::Utility;
+using namespace AppInstaller::Filesystem;
+using namespace TestCommon;
+
+TEST_CASE("PathEscapesDirectory", "[filesystem]")
+{
+    TestCommon::TempDirectory tempDirectory("TempDirectory");
+    const std::filesystem::path& basePath = tempDirectory.GetPath();
+
+    std::string badRelativePath = "../../target.exe";
+    std::string badRelativePath2 = "test/../../target.exe";
+    std::string goodRelativePath = "target.exe";
+    std::string goodRelativePath2 = "test/../test1/target.exe";
+
+    std::filesystem::path badPath = basePath / ConvertToUTF16(badRelativePath);

```

Comment:
Converting to UTF-16 is necessary here because you know that the input is ASCII.